tensorflow/compiler/mlir/hlo/BUILD

```diff
@@ -437,6 +437,42 @@ cc_library(
     alwayslink = 1,
 )
 
+gentbl_cc_library(
+    name = "lhlo_structured_interface_gen",
+    compatible_with = get_compatible_with_cloud(),
+    tbl_outs = [
+        (
+            ["-gen-op-interface-decls"],
+            "include/mlir-hlo/Dialect/mhlo/IR/lhlo_structured_interface.h.inc",
+        ),
+        (
+            ["-gen-op-interface-defs"],
+            "include/mlir-hlo/Dialect/mhlo/IR/lhlo_structured_interface.cpp.inc",
+        ),
+    ],
+    tblgen = "@llvm-project//mlir:mlir-tblgen",
+    td_file = "include/mlir-hlo/Dialect/mhlo/IR/lhlo_structured_interface.td",
+    deps = [":hlo_ops_td_files"],
+)
+
+cc_library(
+    name = "lhlo_structured_interface",
+    srcs = [
+        "lib/Dialect/mhlo/IR/lhlo_structured_interface.cc",
+    ],
+    hdrs = [
+        "include/mlir-hlo/Dialect/mhlo/IR/lhlo_structured_interface.h",
+        "include/mlir-hlo/Dialect/mhlo/IR/lhlo_structured_interface.h.inc",
+    ],
+    includes = ["include"],
+    deps = [
+        ":lhlo_structured_interface_gen",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:Support",
+    ],
+    alwayslink = 1,
```

Comment:
We should never need "alwayslink" anywhere in this codebase, can you remove?

Reply:
I removed the alwayslink option for all the utils, but reserved in all the passes. Please double check. (as I see other passes did the same thing. It's still potential for mlir-hlo repo to be build into a so file, isn't it?)

Reply:
> It's still potential for mlir-hlo repo to be build into a so file, isn't it?

alwayslink shouldn't be needed for this (I hope), are you saying that you can't build .so with Bazel without using alwayslink everywhere?

The issue with alwayslink is that it blows up binary size, the only reason (I know of) it is needed is when you have global constructor / global registration (which we tried to move away from in MLIR last year).